docs/source/links/state.md

```diff
@@ -0,0 +1,295 @@
+`withClientState` and passing in a resolver map. A resolver map describes how to
+retrieve and update your local data.
+
+Let's look at an example where we're using a GraphQL mutation to update whether
+our network is connected with a boolean flag:
+
+```js
+import { withClientState } from 'apollo-link-state';
+
+const stateLink = withClientState({
+  Mutation: {
+    networkStatus: (_, { isConnected }, { cache }) => {
+      const data = {
+        networkStatus: { isConnected, __typename: 'NetworkStatus' },
+      };
+      cache.writeData({ data });
+      return data;
+    },
+  },
+});
+```
+
+To hook up your state link to Apollo Client, concatenate it to the other links
+in your Apollo Link chain. Your state link should be one of the last links in
+your chain. Then, pass your link chain to the Apollo Client constructor.
+
+```js
+const client = new ApolloClient({
+  cache: new InMemoryCache(),
+  link: stateLink.concat(new HttpLink()),
+});
+```
+
+How do we differentiate a request for local data from a request that hits our
+server? In our query or mutation, we specify which fields are client-only with a
+`@client` directive. This tells our network stack to retrieve or update the data
+in the cache with our resolver map that we passed into our state link.
+
+```js
+const UPDATE_NETWORK_STATUS = gql`
+  mutation updateNetworkStatus($isConnected: Boolean) {
+    networkStatus(isConnected: $isConnected) @client {
+      isConnected
+    }
+  }
+`;
+```
+
+To fire off the mutation from your component, bind your mutation to your
+component via your favorite Apollo view layer integration just like you normally
+would. Here's what this would look like for React:
+
+```js
+const WrappedComponent = graphql(UPDATE_NETWORK_STATUS, {
+  props: ({ mutate }) => ({
+    updateNetworkStatus: isConnected => mutate({ variables: { isConnected } }),
+  }),
+})(NetworkStatus);
+```
+
+Now that you've seen how easy it is to update local state with
+`apollo-link-state`, let's dive deeper into the resolver map.
+
+<h2 id="resolver">Resolver map</h2>
+
+Your resolver map is where all the magic happens to retrieve and update your
+local data in the Apollo cache. The resolver map is an object with resolver
+functions for each GraphQL object type. You can think of a GraphQL query or
+mutation as a tree of function calls for each field.
+
+Each field must either resolve to a GraphQL primitive (Int, String, etc) or
+resolve in a function call. You don't have to specify resolver functions for
+every field, however. GraphQL has [default
+resolvers](docs/graphql-tools/resolvers.html#Default-resolver). If the return
+value from the parent object has the same property names as the fields requested
+in the child object, you won't need to specify a resolver.
+
+```js
+const getUser = gql`
+  query {
+    user(id: 1) {
+      name {
+        last
+        first
+      }
+    }
+  }
+`;
+```
+
+For this query, you will need to specify a resolver for `Query.user` in your
```

Comment:
This seems like it has veered into general graphql documentation, especially since I don't see a `@client` directive above.

Reply:
Would adding a `@client` directive help? Idk the right balance to strike here, I want people who aren't too familiar w/ GraphQL or `graphql-tools` to still be able to feel productive right away.

Reply:
I think it might be OK if we focus on a target audience of people who already know graphql a bit, but I think this section is fine. Let's add that client directive tho